src/ios-deploy/ios-deploy.m

```diff
@@ -2229,21 +2225,101 @@ void download_dyld_file(AMDeviceRef device, uint32_t dyld_index,
               });
 
     write_dyld_file(download_path, file_size);
-  
-    CFRelease(download_path);
+
     AMDeviceStopSession(device);
     AMDeviceDisconnect(device);
+    return download_path;
+}
+
+CFStringRef create_dsc_bundle_path_for_device(AMDeviceRef device) {
+    CFStringRef xcode_dev_path = copy_xcode_dev_path();
+
+    AMDeviceConnect(device);
+    CFStringRef device_class = AMDeviceCopyValue(device, 0, CFSTR("DeviceClass"));
+    AMDeviceDisconnect(device);
+
+    CFStringRef platform_name;
+    if (CFStringCompare(CFSTR("AppleTV"), device_class, 0) == kCFCompareEqualTo) {
+        platform_name = CFSTR("AppleTVOS");
+    } else if (CFStringCompare(CFSTR("Watch"), device_class, 0) ==
+               kCFCompareEqualTo) {
+        platform_name = CFSTR("WatchOS");
+    } else {
+        platform_name = CFSTR("iPhoneOS");
+    }
+
+    return CFStringCreateWithFormat(
+        NULL, NULL,
+        CFSTR("%@/Platforms/%@.platform/usr/lib/dsc_extractor.bundle"),
+        xcode_dev_path, platform_name);
+}
+
+typedef int (*extractor_proc)(const char *shared_cache_file_path, const char *extraction_root_path,
```

Comment:
This should probably go in MobileDevice.h with most of the other crazy internal typedefs.

Reply:
I'm hesitant to add it there because this typedef comes from dyld, specifically dsc_extractor.bundle, not the mobiledevice framework so it seems it would be out of place there